keras_cv/layers/object_detection/roi_generator_test.py

```diff
@@ -0,0 +1,178 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv.layers.object_detection.roi_generator import ROIGenerator
+
+
+class ROIGeneratorTest(tf.test.TestCase):
+    def test_singlelevel_singlebatch_roi_ignore_box(self):
+        roi_generator = ROIGenerator("xyxy", nms_iou_threshold_train=0.96)
+        rpn_boxes = tf.constant(
+            [
+                [[0, 0, 10, 10], [0.1, 0.1, 9.9, 9.9], [5, 5, 10, 10], [2, 2, 8, 8]],
+            ]
+        )
+        expected_rois = tf.gather(rpn_boxes, [[1, 3, 2]], batch_dims=1)
+        expected_rois = tf.concat([expected_rois, tf.zeros([1, 1, 4])], axis=1)
+        rpn_boxes = {"2": rpn_boxes}
+        rpn_scores = tf.constant(
+            [
+                [0.6, 0.9, 0.2, 0.3],
+            ]
+        )
+        # selecting the 1st, then 3rd, then 2nd as they don't overlap
+        # 0th box overlaps with 1st box
+        expected_roi_scores = tf.gather(rpn_scores, [[1, 3, 2]], batch_dims=1)
+        expected_roi_scores = tf.concat([expected_roi_scores, tf.zeros([1, 1])], axis=1)
+        rpn_scores = {"2": rpn_scores}
+        rois, roi_scores = roi_generator(rpn_boxes, rpn_scores, training=True)
+        self.assertAllClose(expected_rois, rois)
+        self.assertAllClose(expected_roi_scores, roi_scores)
+        # print(rois)
+        # print(expected_rois)
+        # print(roi_scores)
+        # print(expected_roi_scores)
+
+    def test_singlelevel_singlebatch_roi_all_box(self):
+        roi_generator = ROIGenerator("xyxy", nms_iou_threshold_train=0.97)
```

Comment:
for the 0.97/0.96 iou score, we might need some comment for them, since it need some calculation to come up the IOU score between [0, 0, 10, 10] and [0.1, 0.1, 9.9, 9.9]

Reply:
Done.